### ISSUE TITLE
`homey` CLI WSL2 support

### ISSUE THREAD

**Issue by lucasvdh** [athombv/apps-sdk-issues#293]
_**TLDR;** installing/running Homey apps via CLI works when started from Windows but doesn't work when started from within a WSL Ubuntu container which talks to the same Docker instance._

## Description
I've been using [WSL](https://learn.microsoft.com/en-us/windows/wsl/install), or Windows Subsystem for Linux, for quite some time because it provides access to Linux tools and utilities, integration with Docker, improved compatibility, easy installation and management for software development while keeping your Windows environment. My entire setup revolves around being able to just use the Windows operating system but having a Linux environment that shares the same filesystem in which I can do all my software development.

The `homey` CLI used to work just fine in WSL but since it migrated to installing/running apps via docker this has been broken. I'd very much like it if the `homey` CLI command would start working from within a WSL container again, because the windows terminal and managing things like npm and git to run in Windows is no fun&hellip;

## The issue

When I run the `homey` CLI command from within a WSL Ubuntu container, it's able to find all the available Homeys, but when I try to run an app it fails to connect to the selected Homey.

I'm not sure what exactly fails at the moment but I'm guessing the docker image tries to use an incorrect network interface to try and connect to the Homey. 

Because the installing/running of apps now works via Docker, and because Windows and the WSL container both talk to the same Docker instance, I'd expect that it wouldn't matter from which context this is started.

The error given is:

```text
Could not connect to Homey (ws://172.17.0.1:30000).
```

<details>
<summary><b>Click to see full stack trace</b></summary>
<p>

```shell
✓ Starting `my.app@0.0.1` in a Docker container...
 — Press CTRL+C to quit.
Debugger listening on ws://0.0.0.0:9229/766469bb-5818-47ba-a849-5789676063d0
For help, see: https://nodejs.org/en/docs/inspector
2023-02-27T09:16:29.393Z Could not connect to Homey (ws://172.17.0.1:30000).
2023-02-27T09:16:29.396Z Error: websocket error
    at WS.onError (/node_modules/engine.io-client/lib/transport.js:29:17)
    at WebSocket.ws.onerror (/node_modules/engine.io-client/lib/transports/websocket.js:112:33)
    at WebSocket.onError (/node_modules/ws/lib/event-target.js:140:16)
    at WebSocket.emit (node:events:513:28)
    at ClientRequest.<anonymous> (/node_modules/ws/lib/websocket.js:579:15)
    at ClientRequest.emit (node:events:513:28)
    at Socket.socketErrorListener (node:_http_client:494:9)
    at Socket.emit (node:events:513:28)
    at emitErrorNT (node:internal/streams/destroy:157:8)
    at emitErrorCloseNT (node:internal/streams/destroy:122:3) {
  type: 'TransportError',
  description: ErrorEvent {
    target: WebSocket {
      _events: [Object: null prototype],
      _eventsCount: 4,
      _maxListeners: undefined,
      _binaryType: 'nodebuffer',
      _closeCode: 1006,
      _closeFrameReceived: false,
      _closeFrameSent: false,
      _closeMessage: '',
      _closeTimer: null,
      _extensions: {},
      _protocol: '',
      _readyState: 2,
      _receiver: null,
      _sender: null,
      _socket: null,
      _bufferedAmount: 0,
      _isServer: false,
      _redirects: 0,
      _url: 'ws://172.17.0.1:30000/socket.io/?EIO=4&transport=websocket',
      _req: null,
      [Symbol(kCapture)]: false
    },
    type: 'error',
    message: 'connect ECONNREFUSED 172.17.0.1:30000',
    error: Error: connect ECONNREFUSED 172.17.0.1:30000
        at TCPConnectWrap.afterConnect [as oncomplete] (node:net:1278:16) {
      errno: -111,
      code: 'ECONNREFUSED',
      syscall: 'connect',
      address: '172.17.0.1',
      port: 30000
    }
  }
}
───────────────────────────────────────────────────────
✓ Uninstalling `my.app`...
✓ Removing container...
```

</p>
</details>

If there's anything I can do to configure my WSL container so running Homey apps works again that would also be great. But because I haven't really been able to find any documentation about how this works I haven't been able to debug the issue further.

**Comment by WeeJeWel** [comment]
Thank you, we will dive into this issue.

Do you maybe happen to know what the host's IP address is, from the perspective of the container, inside a WSL environment? (So the replacement of `172.17.0.1`).

**Comment by lucasvdh** [comment]
Good to hear!

The Ubuntu container:
```text
➜ ifconfig
eth0: flags=4163<UP,BROADCAST,RUNNING,MULTICAST>  mtu 1350
        inet 172.25.210.236  netmask 255.255.240.0  broadcast 172.25.223.255
        inet6 fe80::215:5dff:fe4e:4a7c  prefixlen 64  scopeid 0x20<link>
        ether 00:15:5d:4e:4a:7c  txqueuelen 1000  (Ethernet)
        RX packets 32262  bytes 20121712 (20.1 MB)
        RX errors 0  dropped 0  overruns 0  frame 0
        TX packets 9537  bytes 1018368 (1.0 MB)
        TX errors 0  dropped 0 overruns 0  carrier 0  collisions 0
```

Homey app runner:
```text
/homey-app-runner # ifconfig
eth0      Link encap:Ethernet  HWaddr 02:42:AC:11:00:02  
          inet addr:172.17.0.2  Bcast:172.17.255.255  Mask:255.255.0.0
          UP BROADCAST RUNNING MULTICAST  MTU:1500  Metric:1
          RX packets:7 errors:0 dropped:0 overruns:0 frame:0
          TX packets:0 errors:0 dropped:0 overruns:0 carrier:0
          collisions:0 txqueuelen:0
          RX bytes:646 (646.0 B)  TX bytes:0 (0.0 B)
```

**Comment by WeeJeWel** [comment]
Interesting, the IP is the same.

I assume `homey app run` runs on Windows? Then the question is, how to reach the Windows' IP address?

**Comment by WeeJeWel** [comment]
Sorry, I just read you run _inside_ WSL.

Have you tried WSL2? I must apologize, we're not very much Windows users at Athom, so it can be overwhelming to understand the status quo of WSL.

**Comment by lucasvdh** [comment]
No worries. 

I should've stated this in the original issue description; whenever I refer to WSL I'm actually referring to WSL2.

Could you maybe share some insight into what the process inside the homey-app-runner is actually trying to reach when it connects to `ws://172.17.0.1:30000`? Is this some sort of proxy that the `homey` CLI sets up?

Some additional context:

<details>
<summary>These are my Windows network interfaces</summary>
<p>

```text
> ipconfig

Windows IP Configuration


Unknown adapter wg0:

   Connection-specific DNS Suffix  . :
   IPv6 Address. . . . . . . . . . . : fd00::10:97:13
   IPv4 Address. . . . . . . . . . . : 10.99.97.13
   Subnet Mask . . . . . . . . . . . : 255.255.255.0
   Default Gateway . . . . . . . . . :

Ethernet adapter Ethernet 2:

   Connection-specific DNS Suffix  . : localdomain
   Link-local IPv6 Address . . . . . : xxxx:xxxxx:xxxx:xxxx:xxxxxxx
   IPv4 Address. . . . . . . . . . . : 192.168.1.22
   Subnet Mask . . . . . . . . . . . : 255.255.255.0
   Default Gateway . . . . . . . . . : 192.168.1.1

Tunnel adapter Teredo Tunneling Pseudo-Interface:

   Connection-specific DNS Suffix  . :
   IPv6 Address. . . . . . . . . . . : xxxx:x:xxxx:xxx:xxxx:xxxx:xxxx:xxxx
   Link-local IPv6 Address . . . . . : fe80::1c8a:36df:a42f:c3e6%20
   Default Gateway . . . . . . . . . :

Ethernet adapter vEthernet (Default Switch):

   Connection-specific DNS Suffix  . :
   Link-local IPv6 Address . . . . . : xxxx::xxxx:xxxx:xxxx:xxxxxxx
   IPv4 Address. . . . . . . . . . . : 172.17.48.1
   Subnet Mask . . . . . . . . . . . : 255.255.240.0
   Default Gateway . . . . . . . . . :

Ethernet adapter vEthernet (WSL):

   Connection-specific DNS Suffix  . :
   Link-local IPv6 Address . . . . . : xxxx:xxxxx:xxxx:xxxx:xxxxxxx
   IPv4 Address. . . . . . . . . . . : 172.25.208.1
   Subnet Mask . . . . . . . . . . . : 255.255.240.0
   Default Gateway . . . . . . . . . :

```
</p>
</details>

**Comment by WeeJeWel** [comment]
Thank you.

`homey app run` starts a webserver, that the Docker container should connect to. That's it, really.

**Comment by jeroenwienk** [comment]
I'm assuming 172.17.0.1 is Windows host and not the Ubuntu host where `homey app run` started the server. I'm not sure why your Ubuntu container is using the hosts Docker.

**Comment by lucasvdh** [comment]
`172.17.0.1` is from the `bridge` Docker network, so yeah I'd also assume that's my Windows host.

---

> I'm not sure why your Ubuntu container is using the hosts Docker.

Do you mean why the docker client in my Ubuntu VM connects to the docker instance of my Windows host? If so: thats a setting I've enabled.

<img src="https://user-images.githubusercontent.com/4311594/221604786-46e14fd6-2259-4fdb-b618-c6371c31bc10.png" width=400/>

I use my Ubuntu VM that runs inside WSL as a place where I host all my software development tools because the Windows terminal and managing things like npm and git to run in Windows is no fun.

So this is where all my software projects are located and also where some dependencies like node, python, php, etc are installed for projects that don't run in docker. A lot of people who develop on Windows but don't necessarily need the Windows terminal have come to adopt this method.

**Comment by WeeJeWel** [comment]
I'll see if I can reproduce/fix this tomorrow, when I have access to a Windows PC :-)

**Comment by WeeJeWel** [comment]
This should now be solved.

```
$ npm i -g homey@beta
```

Can you confirm this works for you?

**Comment by lucasvdh** [comment]
Works perfectly! 

Thanks for the quick fix guys 😄
